dafni/Dockerfile

```diff
@@ -0,0 +1,32 @@
+# Define the Python version neded for CTF
+FROM python:3.10-slim
+
+## Prevents Python from writing pyc files
+ENV PYTHONDONTWRITEBYTECODE=1
+#
+## Keeps Python from buffering stdout and stderr to avoid the framework
+## from crashing without emitting any logs due to buffering
+ENV PYTHONUNBUFFERED=1
+
+#Label maintainer
+LABEL maintainer="Dr. Farhad Allian - The University of Sheffield"
+
+# Copy the source code and test files from build into the container
+COPY --chown=nobody ./causal_testing /usr/src/app/
+COPY --chown=nobody ./dafni/inputs /usr/src/app/inputs/
+COPY --chown=nobody ./dafni/main_dafni.py /usr/src/app/
+
+# Change the working directory
+WORKDIR /usr/src/app/
+
+# Install core dependencies using PyPi
+RUN pip install causal-testing-framework --no-cache-dir
```

Comment:
Out of curiosity why is the `--no-cache-dir` flag used here? My guess would be that that there is no `causal-testing-framework` wheel in cache as it's a fresh container.

Reply:
@cwild-UoS The ``--no-cache-dir`` option disables the downloading and storing of cached packages, which reduces the overall Docker image size. I haven't calculated what that difference is, but it's not too important for our purposes

Reply:
Makes lots of sense! Good idea